integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/DistroXClusterFaultTolerantUpscaleTest.java

```diff
@@ -0,0 +1,233 @@
+import com.sequenceiq.it.cloudbreak.dto.distrox.image.DistroXImageTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.instancegroup.DistroXNetworkTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.testcase.mock.clouderamanager.AbstractClouderaManagerTest;
+
+public class DistroXClusterFaultTolerantUpscaleTest extends AbstractClouderaManagerTest {
+
+    private static final int SUCCESSFUL_UPSCALE_NODE_COUNT = 10;
+
+    private static final int FAILING_NODE_COUNT = 2;
+
+    private static final String UPSCALED_INSTANCE_GROUP_NAME = "worker";
+
+    private static final String IMAGE_CATALOG_ID = "f6e778fc-7f17-4535-9021-515351df3691";
+
+    private static final String CM_FOR_DISTRO_X = "cm4dstrx";
+
+    private static final String CLUSTER_KEY = "cmdistrox";
+
+    private static final String DIX_IMG_KEY = "dixImg";
+
+    private static final String DIX_NET_KEY = "dixNet";
+
+    private static final Duration POLLING_INTERVAL = Duration.of(3000, ChronoUnit.MILLIS);
+
+    @Inject
+    private BlueprintTestClient blueprintTestClient;
+
+    @Inject
+    private DistroXTestClient distroXClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "there is a running DistroX cluster",
+            when = "an upscale is called, but some of the new nodes are unreachable",
+            then = "the cluster should be AVAILABLE in a partially upscaled state")
+    public void testUpscaleWithFewNodesBeingUnreachable(MockedTestContext testContext, ITestContext testNgContext) {
+        DistroXStartStopTestParameters params = new DistroXStartStopTestParameters(testNgContext.getCurrentXmlTest().getAllParameters());
+        String stack = resourcePropertyProvider().getName();
+        createDatalake(testContext);
+
+        MinionIpAddressesResponse minionIpAddressesResponse = getMockResponseForIpAddrsSaltCall();
+        Set<String> saltBodyFilters = getSaltBodyFilters();
+
+        DistroXTestDto currentContext = testContext
+                .given(DIX_NET_KEY, DistroXNetworkTestDto.class)
+                .given(DIX_IMG_KEY, DistroXImageTestDto.class)
+                .withImageCatalog()
+                .withImageId(IMAGE_CATALOG_ID)
+                .given(CM_FOR_DISTRO_X, DistroXClouderaManagerTestDto.class)
+                .given(CLUSTER_KEY, DistroXClusterTestDto.class)
+                .withValidateBlueprint(false)
+                .withClouderaManager(CM_FOR_DISTRO_X)
+                .given(stack, DistroXTestDto.class)
+                .withCluster(CLUSTER_KEY)
+                .withName(stack)
+                .withImageSettings(DIX_IMG_KEY)
+                .withNetwork(DIX_NET_KEY)
+                .when(distroXClient.create(), key(stack))
+                .await(STACK_AVAILABLE, key(stack).withIgnoredStatues(Set.of(Status.UNREACHABLE)))
```

Comment:
this code part can be extracted into a private method, since both test using this same setup